week5.4-ex1.py

```diff
@@ -0,0 +1,29 @@
+import itertools
+from typing import Iterable
+
+
```

Comment:
Saw your comment, the code is indeed very nice :)

Reply:
Greate:) 
I see you approved the changes, it means I can merge this branch with the main?